addons/account_edi_proxy_client/models/account_edi_proxy_user.py

```diff
@@ -166,13 +169,19 @@ def _renew_token(self):
         that multiple database use the same credentials. When receiving an error for an expired refresh_token,
         This method makes a request to get a new refresh token.
         '''
+        try:
+            with self.env.cr.savepoint(flush=False):
+                self.env.cr.execute('SELECT * FROM account_edi_proxy_client_user WHERE id IN %s FOR UPDATE NOWAIT', [tuple(self.ids)])
```

Comment:
Is it the right thing to lock? Can' there be multiple records of proxy clients for one invoice, for instance?

Reply:
Most likely not

Reply:
I think it is the right thing to lock because it is on that record that the refresh_token is updated.